packages/webcomponents/src/components/payment-method-form/test/payment-method-form.spec.tsx

```diff
@@ -17,79 +26,209 @@ describe('justifi-payment-method-form', () => {
+    it('removes message listener on disconnected callback', async () => {
+      const removeSpy = jest.spyOn(FrameCommunicationService.prototype, 'removeMessageListener');
+
+      const page = await newSpecPage({
+        components: [PaymentMethodForm],
+        template: () => <justifi-payment-method-form iframeOrigin={config.iframeOrigin}></justifi-payment-method-form>,
+      });
+
+      // Manually set up the frameService to ensure it's available for the disconnectedCallback
+      page.rootInstance.frameService = new FrameCommunicationService(page.rootInstance.iframeElement, config.iframeOrigin);
+      page.rootInstance.frameService.addMessageListener(jest.fn()); // Add a dummy listener
+
+      // Manually call disconnectedCallback to simulate the component being removed
+      await page.rootInstance.disconnectedCallback();
+
+      // Check if removeMessageListener was called upon disconnection
+      expect(removeSpy).toHaveBeenCalled();
+
+      // Cleanup the spy
+      removeSpy.mockRestore();
+    });
   });
 
-  it('renders a card iframe with singleLine when paymentMethodFormType is card and singleLine is set', async () => {
-    const page = await newSpecPage({
-      components: [PaymentMethodForm],
-      template: () => <justifi-payment-method-form paymentMethodFormType='card' singleLine={true}></justifi-payment-method-form>,
+  // Event emission tests
+  describe('Event Emissions', () => {
+    it('emits paymentMethodFormReady when iframe sends ready event', async () => {
+      const page = await newSpecPage({
+        components: [PaymentMethodForm],
+        template: () => (
+          <justifi-payment-method-form
+            paymentMethodFormType={PaymentMethodTypes.card}
+            iframeOrigin={config.iframeOrigin}
+          ></justifi-payment-method-form>
+        ),
+      });
+
+      await page.waitForChanges();
+
+      const eventSpy = jest.fn();
+      page.win.addEventListener('paymentMethodFormReady', eventSpy);
+
+      // Ensure paymentMethodFormType is set and is a valid key in MessageEventType
+      const formType = page.rootInstance.paymentMethodFormType;
+      const eventType = MessageEventType[formType] ? MessageEventType[formType].ready : null;
+
+      if (!eventType) {
+        throw new Error(`Invalid paymentMethodFormType: ${formType}`);
+      }
+
+      const simulatedEventData = { eventType, data: {} };
+
+      page.rootInstance.dispatchMessageEvent({ data: simulatedEventData } as MessageEvent);
+      await page.waitForChanges();
+
+      expect(eventSpy).toHaveBeenCalled();
     });
+  });
 
-    expect(page.root).toEqualHtml(`
-      <justifi-payment-method-form>
-        <iframe id='justifi-payment-method-form-card' src='${config.iframeOrigin}/card?singleLine=true'></iframe>
-      </justifi-payment-method-form>
-    `);
+  describe('Methods', () => {
+    it('calls postMessageWithResponseListener with correct parameters on tokenize', async () => {
```

Comment:
Great work here with these new tests! These are so helpful, and readable as well!